uninstall.php

```diff
@@ -8,39 +8,133 @@
  * @package Airstory
  */
 
-namespace Airstory;
+namespace Airstory\Uninstall;
+
+use Airstory\Connection as Connection;
 
 require_once __DIR__ . '/includes/connection.php';
 
-global $wpdb;
+/**
+ * Retrieve the IDs of any sites with active Airstory connections.
+ *
+ * @global $wpdb
+ *
+ * @return array An array of WordPress site IDs that have one or more active Airstory connections.
+ */
+function get_active_site_ids() {
+	global $wpdb;
+
+	/*
+	 * The SQL query attempts to parse a site ID out of the key name within wp_usermeta.
+	 *
+	 * Assuming the default WordPress $table_prefix (as set in wp-config.php) of "wp_", keys will
+	 * look like:
+	 *
+	 * - wp__airstory_target
+	 * - wp_2__airstory_target
+	 * - wp_3__airstory_target
+	 *
+	 * The SUBSTRING() function within MySQL accepts three arguments: the string, the starting point
+	 * (starting at 0), and the length of the string to return.
+	 *
+	 * The meta_key will be our string, and the length of $table_prefix (available via the
+	 * $wpdb->base_prefix property) + 1 gives us a starting point (e.g. everything before the first
+	 * digit of the site ID).
+	 *
+	 * The third argument is where things get more complicated: we need the length of the meta_key,
+	 * less the length of the static portion ("_airstory_target"), the length of the table prefix,
+	 * and one more (for the extra underscore immediately following the site ID).
+	 */
+	$site_ids = $wpdb->get_col( $wpdb->prepare( "
+		SELECT DISTINCT SUBSTRING(
+			meta_key,
+			LENGTH(%s) + 1,
+			LENGTH(meta_key) - LENGTH('_airstory_target') - LENGTH(%s) - 1
+		) AS site_id
+		FROM $wpdb->usermeta WHERE meta_key LIKE '%_airstory_target'
+		ORDER BY site_id;", $wpdb->base_prefix, $wpdb->base_prefix ) );
+	$site_ids = array_map( 'intval', $site_ids );
+
+	return array_values( array_filter( $site_ids ) );
+}
+
+/**
+ * Collect any users that are connected to Airstory and close their connections.
+ */
+function disconnect_all_users() {
+	$user_args       = array(
+		'fields'      => 'ID',
+		'number'      => 100,
+		'count_total' => false,
+		'meta_query'  => array(
+			array(
+				'key'     => '_airstory_target',
+				'compare' => 'EXISTS',
+			),
+		),
+	);
+	$connected_users = new \WP_User_Query( $user_args );
+	$user_ids        = $connected_users->results;
+
+	while ( ! empty( $user_ids ) ) {
+		$user_id = array_shift( $user_ids );
+
+		Connection\remove_connection( $user_id );
+
+		// If we've reached the end, get the next batch.
+		if ( empty( $user_ids ) ) {
+			$connected_users = new \WP_User_Query( $user_args );
+			$user_ids        = $connected_users->results;
+		}
+	}
+}
```

Comment:
The while loop effectively batches these into groups of 100; we copy the `WP_User_Query::$results` property to `$user_ids` and treat that as a queue, disconnecting one connection at a time. When `$user_ids` is empty, we run the same query again, which will give us up to 100 new `_airstory_target`-having users (I've checked, and `WP_User_Query` doesn't seem to cache its results).